Automattic/jetpack
# CSSTidy: preserve leading decimal zeros in Gutenberg

Fixes https://github.com/Automattic/wp-calypso/issues/45629

## Changes proposed in this Pull Request:

CSSTidy performs some "optimizations" that trigger Gutenberg block validation errors. 

To prevent this, we've had to add some "optimizations" of our own to CSSTidy's inner workings. See, for example, 
https://github.com/Automattic/jetpack/pull/19935

This PR adds a setting to CSSTidy to tell its optimizer to preserve leading decimal zeros.

Currently, CSSTidy strips the `0` from `0.7` resulting in `.7`.

For Gutenberg inline styles, `0.7` is a valid CSS value for line-height for example. Where the post body content contains `.7`, we see block validation errors 😱 

<img width="723" alt="Screen Shot 2021-06-01 at 2 56 05 pm" src="https://user-images.githubusercontent.com/6458278/120269326-ef6d8500-c2ea-11eb-95bb-0fe84cf12eb0.png">

### TODO
- [x] Fix linting issues 😸 

## Does this pull request change what data or activity we track or use?
Nope.

## Testing instructions:
Patch your sandbox with the diff created by this PR

Create a paragraph or heading block and select a line-height value of less than `0`

```html

<p style="line-height:0.4">AM I VALID? AM I?</p>

```

Save the post and refresh the page.

You should not see any validation errors, e.g., 

```html
Content generated by `save` function:

<p>AM I VALID? AM I?</p>

Content retrieved from post body:

<p style="line-height:.4;">AM I VALID? AM I?</p>
```

### Unit tests

Run `yarn docker:up` from the Jetpack repo root directory.

In another terminal run `yarn run docker:phpunit --testsuite csstidy`.
